viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
+
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    """- add the `movie` to the `"watched"` list inside of `user_data`
+- return the `user_data`"""
+    user_data = {"title":title,"genre":genre,"rating":rating}
+    if title and genre and rating:
+        return user_data
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    """- add the `movie` to the `"watched"` list inside of `user_data`
+- return the `user_data`"""
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """- add the `movie` to the `"watchlist"` list inside of `user_data`
+- return the `user_data`"""
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):
+    """- If the title is in a movie in the user's watchlist:
+- remove that movie from the watchlist
+- add that movie to watched
+- return the `user_data`
+- If the title is not a movie in the user's watchlist:
+- return the `user_data`"""
+    
+    for i in user_data["watchlist"]:
+        if i["title"] == title:
+            user_data["watchlist"].remove(i)
+            user_data["watched"].append(title)
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    """- Calculate the average rating of all movies in the watched list
+- The average rating of an empty watched list is `0.0`
+- return the average rating"""
+    total = 0
+    avg_rating = 0
+    if user_data["watched"]:
+        for i in user_data["watched"]:
+            total += i["rating"]
+        avg_rating = total/len(user_data["watched"])
+    
+        return avg_rating
+    else: 
+    
+        return 0.0
+
+def get_most_watched_genre(user_data):
+    """- Determine which genre is most frequently occurring in the watched list
+- return the genre that is the most frequently watched
+- If the value of "watched" is an empty list, `get_most_watched_genre` should return `None`."""
+
+    list_genre = []
+    if user_data["watched"]:
+        for i in user_data["watched"]:
+            list_genre.append(i["genre"])
+        rpt_genre = max(set(list_genre), key=list_genre.count)
+    
+        return rpt_genre
+    
+    else: 
+    
+        return None
 
 # -----------------------------------------
-# ------------- WAVE 3 --------------------
 # -----------------------------------------
+# ------------- WAVE 3 --------------------
 
-        
+def helper_join_list_of_lists_and_removes_duplicates(list_of_lists):
+    plain_list = []
+    for sublist in list_of_lists:
+        for item in sublist:
+            plain_list.append(item)
+    
+
+    list_of_dictionaries_comp= [dict(s) for s in set(frozenset(myObject.items()) for myObject in plain_list)]

```

Comment:
This is another good example of where an explanatory comment would help clarify your understanding of this complex list comprehension.